mobile/src/navigation/app/helpers/get-allowed-screens.helper.ts

```diff
@@ -0,0 +1,20 @@
+import { DrawerNavigationItem, PermissionItem } from '~/common/types/types';
+
+const getAllowedScreens = (
+  screens: DrawerNavigationItem[],
+  userPermissions: PermissionItem[],
+): DrawerNavigationItem[] => {
+  const userPermissionKeys = userPermissions.map((item) => item.key);
+
+  return screens.filter((screen) => {
+    if (!screen.permissions.length) {
+      return true;
+    }
+
+    return screen.permissions.some((permission) =>
+      userPermissionKeys.includes(permission),
+    );
```

Comment:
We should return screen only if user has all the permissions
```suggestion
    return screen.permissions.every((permission) =>
      userPermissionKeys.includes(permission),
    );
```

Reply:
I misunderstood. I thought there must be at least one